test/Parse/try_swift4.swift

```diff
@@ -0,0 +1,260 @@
+// RUN: %target-typecheck-verify-swift -swift-version 4
```

Comment:
Even though git is showing this as a new file, this just the existing `try.swift` test with a few new tests added to differentiate between the swift4 and the swift 5 behavior..